osu.Game/Overlays/SkinEditor/SkinEditor.cs

```diff
@@ -361,6 +406,54 @@ private void revert()
             }
         }
 
+        protected void Cut()
+        {
+            Copy();
+            DeleteItems(SelectedComponents.ToArray());
+        }
+
+        protected void Copy()
+        {
+            clipboard.Content.Value = JsonConvert.SerializeObject(SelectedComponents.Cast<Drawable>().Select(s => s.CreateSerialisedInfo()).ToArray());
+        }
+
+        protected void Clone()
+        {
+            // Avoid attempting to clone if copying is not available (as it may result in pasting something unexpected).
+            if (!canCopy.Value)
+                return;
+
+            // This is an initial implementation just to get an idea of how people used this function.
+            // There are a couple of differences from osu!stable's implementation which will require more work to match:
+            // - The "clipboard" is not populated during the duplication process.
+            // - The duplicated hitobjects are inserted after the original pattern (add one beat_length and then quantize using beat snap).
+            // - The duplicated hitobjects are selected (but this is also applied for all paste operations so should be changed there).
+            Copy();
+            Paste();
+        }
+
+        protected void Paste()
+        {
+            changeHandler?.BeginChange();
+
+            var drawableInfo = JsonConvert.DeserializeObject<SerialisedDrawableInfo[]>(clipboard.Content.Value);
+
+            if (drawableInfo == null)
+                return;
+
+            var instances = drawableInfo.Select(d => d.CreateInstance())
+                                        .OfType<ISerialisableDrawable>()
+                                        .ToArray();
+
+            foreach (var i in instances)
+                placeComponent(i, false);
```

Comment:
There are zero protections that would check whether what is being pasted actually *can* be pasted onto the screen, so you can easily crash your game via the following:

1. Enter gameplay
2. Open skin editor
3. Copy a HUD element that depends on gameplay state
4. Move to song select, open skin editor
5. Paste to :boom: 

https://user-images.githubusercontent.com/20418176/219803189-e6e630ff-7c95-4935-ae81-d0497b61b955.mp4

Cause is the clipboard being shared at `SkinEditorOverlay` level combined with lack of examination of pasted content in a manner similar to what `SkinComponentToolbox` is doing.

Reply:
I meant to fix this one yesterday (noticed while working on one of the other ten branches) but it slipped my mind 😅

Reply:
I've fixed this by making `placeComponent` gracefully handle errors, which seems in line with what we're doing elsewhere with skin element handling.